Mixed Models updates

FBartos opened the pull request:
Features:
- ability to remove random intercept

fixes: https://github.com/jasp-stats/INTERNAL-jasp/issues/1161

Kucharssim commented on R/MixedModelsCommon.R:
```diff
@@ -1765,6 +1844,8 @@ gettextf <- function(fmt, ..., domain = NULL)  {
         estimate =  emmContrast[i, ncol(emmContrast) - 4],
         se       =  emmContrast[i, "SE"],
         df       =  emmContrast[i, "df"],
+        lowerCI  =  emmContrast[i, ncol(emmContrast) - 4] + stats::qt(0.025, df = emmContrast[i, "df"]) * emmContrast[i, "SE"],
```
should we use instead of hard-coded `0.025` and `0.975` use user defined values here?

Kucharssim reviewed (changes requested):
Code looks good, one code suggestion, some specific comments regarding the functionality:

- [ ] https://github.com/jasp-stats/INTERNAL-jasp/issues/1161
just one little nitpick: the component now says "Random *slopes* by [random effects group factor]". Should it now be something like "Random *coefficients*", or something like that?
- [ ] https://github.com/jasp-stats/jasp-issues/issues/1028
something very weird happens here. When I click the "Random effects estimates" button, the analysis does not do anything, but somehow the analysis title disappears. After that, toggling the button or refreshing the analysis does not do anything. Attempting to close the file crashes JASP. Attached jasp file [bug.jasp.zip](https://github.com/jasp-stats/jaspMixedModels/files/9505355/bug.jasp.zip). The first analysis is the button unticked, the second is with the button ticked.
- [x] https://github.com/jasp-stats/jasp-issues/issues/1614
seems to work with the jasp file attached in the issue, so I assume it's fixed
- [ ] https://github.com/jasp-stats/jasp-issues/issues/1012
once we settle the values used for calculating the CIs (see code suggestion), it's fine

FBartos commented:
1) renamed to "Random coefficients (XXX)"
2) Don figured out that there was some type conversion bug (number as string -> number) which was crashing JASP. I fixed the formatting and the functionality now works. (I also found a bug in B(G)LMM with them and added a unit test)
3) The error was correct - the GLMM specification required a proportion of successes. I now extended it to also allow the number of successes (we calculate the proportion internally).
4) Yeah, true -- especially we were already changing the title.

Kucharssim reviewed (approved):
ok, looks good now, the issues I found were fixed as well!

Kucharssim commented:
Hm, actually, why did not the tests run here? Any ideas @vandenman?

vandenman commented:
no idea. @FBartos perhaps add a meaningless change to an R file to trigger the tests?

FBartos commented:
done :)
